### PR TITLE
sbt-ci-deploy.bsh

### PR DESCRIPTION
**Pull request by wrouesnel**
sbt-ci-deploy is lib-ci based sbt publication script. It rationalizes and generalizes some of the contained logic.

### CODE REVIEW

File: t/test-sbt-ci-setup-version.bsh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+tmpdir=$(mktemp -d -p .)
+echo "version in ThisBuild := 1.0.0" > $tmpdir/version.sbt
+cd $tmpdir
+
+versionOrig=$(cat version.sbt)
+# Check command works
+WVPASS ../sbt-ci-setup-version.bsh
+# Check command modifies version
+WVPASSNE "$(cat version.sbt)" "$versionOrig"
```
Comment (stephanh) [inline t/test-sbt-ci-setup-version.bsh]: Shouldn't we check that the change meets our expectations?

Comment (wrouesnel) [inline t/test-sbt-ci-setup-version.bsh]: We run that test in `lib-ci` for Get_Version. Here the only valid thing to check is that we modify the file really.

Comment (stephanh) [inline t/test-sbt-ci-setup-version.bsh]: We need to check that we modify the file as expected, i.e. parse the version number correctly and write the correct string to the file which is not just the version number.

Comment (wrouesnel) [inline t/test-sbt-ci-setup-version.bsh]: Handled.

Comment (stephanh) [inline t/test-sbt-ci-setup-version.bsh]: Where is the positive test? I can't see a test that checks that the content of version.sbt is `version in ThisBuild=....`?

File: t/test-sbt-ci-setup-version.bsh [hunk 2]
```diff
@@ -7,11 +7,15 @@ tmpdir=$(mktemp -d -p .)
...
 
+. ./lib-ci
+
 versionOrig=$(cat version.sbt)
 # Check command works
 WVPASS ../sbt-ci-setup-version.bsh
 # Check command modifies version
 WVPASSNE "$(cat version.sbt)" "$versionOrig"
+# Check input string comes out correctly.
+WVPASS grep -o "version in ThisBuild := $(Version_Get 1.0.0)" version.sbt
```
Comment (wrouesnel) [inline t/test-sbt-ci-setup-version.bsh]: Here.

**Comment by wrouesnel** [comment]
Rebased. The test is in commit 2.

File: t/test-sbt-ci-setup-version.bsh [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash -u
+
+# Test framework
+. ./wvtest.sh
+
+tmpdir=$(mktemp -d -p .)
+echo "version in ThisBuild := 1.0.0" > $tmpdir/version.sbt
```
Comment (stephanh) [inline t/test-sbt-ci-setup-version.bsh]: The version string needs to be surrounded by `"`. See https://github.com/CommBank/maestro/blob/master/version.sbt#L15

File: t/test-sbt-ci-setup-version.bsh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+. ./lib-ci
+
+versionOrig=$(cat version.sbt)
+# Check command works
+WVPASS ../sbt-ci-setup-version.bsh
+# Check command modifies version
+WVPASSNE "$(cat version.sbt)" "$versionOrig"
+# Check input string comes out correctly.
+WVPASS grep -o "version in ThisBuild := $(Version_Get 1.0.0)" version.sbt
```
Comment (stephanh) [inline t/test-sbt-ci-setup-version.bsh]: See comment above about having the version string encased in `"`.

Also can we check that the file doesn't contain anything else, as in that the command overwrites the version file to move the `uniqueVersionSettings` or the `localVersionSettings` which is still to come.

File: tests/test-lib-ci.bsh [hunk 5]
```diff
@@ -67,3 +67,6 @@ export ${CI_SYSTEM}_PULL_REQUEST=false
 unset RELEASE_BRANCHES
 echo "IS_RELEASE=$(Is_Release)"
 WVPASS [ $(Is_Release) == 1 ]
+
+# Test Hostname_From_Url
+WVPASS [ $(Hostname_From_Url "http://test/some/args?hello") = "test" ]
```
Comment (stephanh) [inline tests/test-lib-ci.bsh]: One for https and with subdomains as well?

**Comment by stephanh** [comment]
Happy for you to merge once the last concerns are dealt with. 👍

**Comment by wrouesnel** [comment]
Ok that should cover off all the test cases. I'm going to merge once Travis says okay.

**Pull request merged**